From 70c130ad653972f108ebc08777d7ce3aac789ac7 Mon Sep 17 00:00:00 2001
From: Samuel Dionne-Riel <samuel@dionne-riel.com>
Date: Sat, 9 Jan 2021 02:26:49 -0500
Subject: [PATCH] [mobile-nixos]: exynos/decon_7880: Adds and sets ARGB as
 default

This fixes the issue of Red and Blue channels being mixed up from the
expected.
---
 drivers/video/fbdev/exynos/decon_7880/decon_core.c | 11 ++++++-----
 drivers/video/fbdev/exynos/decon_7880/decon_dsi.c  |  2 +-
 2 files changed, 7 insertions(+), 6 deletions(-)

diff --git a/drivers/video/fbdev/exynos/decon_7880/decon_core.c b/drivers/video/fbdev/exynos/decon_7880/decon_core.c
index 543efe8c41c..95d27143767 100755
--- a/drivers/video/fbdev/exynos/decon_7880/decon_core.c
+++ b/drivers/video/fbdev/exynos/decon_7880/decon_core.c
@@ -1598,16 +1598,17 @@ static u32 get_vpp_out_format(u32 format)
 	case DECON_PIXEL_FORMAT_YVU420:
 	case DECON_PIXEL_FORMAT_YUV420M:
 	case DECON_PIXEL_FORMAT_YVU420M:
-	case DECON_PIXEL_FORMAT_RGB_565:
 	case DECON_PIXEL_FORMAT_BGRA_8888:
-	case DECON_PIXEL_FORMAT_RGBA_8888:
 	case DECON_PIXEL_FORMAT_ABGR_8888:
-	case DECON_PIXEL_FORMAT_ARGB_8888:
 	case DECON_PIXEL_FORMAT_BGRX_8888:
-	case DECON_PIXEL_FORMAT_RGBX_8888:
 	case DECON_PIXEL_FORMAT_XBGR_8888:
-	case DECON_PIXEL_FORMAT_XRGB_8888:
 		return DECON_PIXEL_FORMAT_BGRA_8888;
+	case DECON_PIXEL_FORMAT_RGB_565:
+	case DECON_PIXEL_FORMAT_RGBA_8888:
+	case DECON_PIXEL_FORMAT_ARGB_8888:
+	case DECON_PIXEL_FORMAT_RGBX_8888:
+	case DECON_PIXEL_FORMAT_XRGB_8888:
+		return DECON_PIXEL_FORMAT_RGBA_8888;
 	default:
 		return format;
 	}
diff --git a/drivers/video/fbdev/exynos/decon_7880/decon_dsi.c b/drivers/video/fbdev/exynos/decon_7880/decon_dsi.c
index 961fbb768d2..77902346e43 100755
--- a/drivers/video/fbdev/exynos/decon_7880/decon_dsi.c
+++ b/drivers/video/fbdev/exynos/decon_7880/decon_dsi.c
@@ -356,7 +356,7 @@ int decon_pan_display(struct fb_var_screeninfo *var, struct fb_info *info)
 		break;
 	case 24:
 	case 32:
-		config.format = DECON_PIXEL_FORMAT_ABGR_8888;
+		config.format = DECON_PIXEL_FORMAT_ARGB_8888;
 		shift = 4;
 		break;
 	default:
-- 
2.29.2

